internal/services/automation/automation_software_update_configuration.go

```diff
@@ -0,0 +1,952 @@
+				string(automation.OperatingSystemTypeWindows),
+			}, false),
+		},
+
+		"linux": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+
+					"reboot_setting": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+					},
+
+					"classification": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+						ValidateFunc: validation.StringInSlice(func() (vs []string) {
+							for _, v := range automation.PossibleLinuxUpdateClassesValues() {
+								vs = append(vs, string(v))
+							}
+							return
+						}(), false),
+					},
+
+					"excluded_packages": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Elem: &pluginsdk.Schema{
+							Type:         pluginsdk.TypeString,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+					},
+
+					"included_packages": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Elem: &pluginsdk.Schema{
+							Type:         pluginsdk.TypeString,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+					},
+				},
+			},
+		},
+
+		"windows": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+
+					"classification": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+						ValidateFunc: validation.StringInSlice(func() (vs []string) {
+							for _, v := range automation.PossibleWindowsUpdateClassesValues() {
+								vs = append(vs, string(v))
+							}
+							return
+						}(), false),
+					},
+
+					"excluded_knowledge_base_numbers": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Elem: &pluginsdk.Schema{
+							Type:         pluginsdk.TypeString,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+					},
+
+					"included_knowledge_base_numbers": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Elem: &pluginsdk.Schema{
+							Type:         pluginsdk.TypeString,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+					},
+
+					"reboot_setting": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+					},
+				},
+			},
+		},
+
+		"duration": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ValidateFunc: validation.StringMatch(regexp.MustCompile(`^PT\d+H\d+M\d+S$`),
+				"Duration needs to be specified using the format PT[n]H[n]M[n]S as per ISO8601"),
+		},
+
+		"virtual_machines": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Schema{
+				Type:         pluginsdk.TypeString,
+				ValidateFunc: computeValidate.VirtualMachineID,
+			},
+		},
+
+		"non_azure_computer_names": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Schema{
+				Type:         pluginsdk.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+
+		"targets": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
```

Comment:
can you have multiple targets blocks? seems a bit weird to have a list of lists?

Reply:
Yes it should limit MaxItem to 1